packages/go_router/lib/src/delegate.dart

```diff
@@ -65,18 +98,18 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       pageKey: pageKey,
     );
 
-    _matches.push(newPageKeyMatch);
+    _matchList.push(newPageKeyMatch);
     notifyListeners();
   }
 
   /// Returns `true` if there is more than 1 page on the stack.
   bool canPop() {
-    return _matches.canPop();
+    return _matchList.canPop();
```

Comment:
This also need to take into account of shell route navigator, the logic should be similar to popRoute